openedx/features/partners/forms.py

```diff
@@ -12,3 +25,129 @@ def clean_email(self):
         self.error_messages['unknown'] = _("We don't recognize the email: {}").format(self.cleaned_data["email"])
         return super(PartnerResetPasswordForm, self).clean_email()
 
+
+class UsernameField(student_forms.UsernameField):
+    """
+    Inheriting UsernameField from student app. Parent class is not validating username already exists
+    """
+
+    def clean(self, value):
+        clean_username = super(UsernameField, self).clean(value)
+
+        # check if user name is not already taken
+        conflicts = check_account_exists(username=clean_username)
+        if conflicts and 'username' in conflicts:
+            raise forms.ValidationError(
+                "The username you entered is already being used. Please enter another username.")
+
+        return clean_username
+
+
+class PartnerAccountCreationForm(forms.Form):
+    """
+    A form to validate account creation data for Give2Asia. It is currently only used for
+    validation, not rendering.
+    """
+
+    _NAME_MIN_LENGTH = 3
```

Comment:
as far as it is encapsulated within the class there was no need to use prefix `_`.

Reply:
I moved this file from `partner/give2asia/forms` to `partner/forms` as the generalisation part of the ticket, and the changes are not mine. Also, I think we should keep it like this [because](https://pep8.org/#descriptive-naming-styles) even though Python does not enforce the scope, it is still a good hint for an internally used variable.